src/Encamina.Enmarcha.SemanticKernel/Extensions/KernelExtensions.cs

```diff
@@ -314,7 +314,7 @@ private static string GetResourceNameFromPluginInfoByFileName(IGrouping<(string
     private static string ReadResource(Assembly assembly, string resourceName)
     {
         using var stream = assembly.GetManifestResourceStream(resourceName);
-        using var streamReader = new StreamReader(stream);
+        using var streamReader = new StreamReader(stream!);
```

Comment:
It seems that `GetManifestResourceStream` can return null. Maybe we can throw an explanative exception in that case.

```suggestion
    // Check if the stream is null and throw an exception if it is
    if (stream is null)
    {
        throw new ArgumentNullException(nameof(stream), "The resource stream is null. Ensure the resource name is correct and the resource is embedded in the assembly.");
    }

        using var streamReader = new StreamReader(stream!);
```